src/installer.ts

```diff
@@ -21,15 +21,52 @@ function voltaVersionHasSetup(version: string): boolean {
   return semver.gte(version, '0.7.0');
 }
 
-export function buildDownloadUrl(platform: string, version: string): string {
+async function getOpenSSLVersion(version = ''): Promise<string> {
+  if (version === '') {
+    let output = '';
+    const options: ExecOptions = {};
+    options.listeners = {
+      stdout: (data: Buffer) => {
+        output += data.toString();
+      },
+      stderr: (data: Buffer) => {
+        output += data.toString();
+      },
+    };
+
+    await exec('openssl version', [], options);
+
+    version = output;
+  }
+
+  // typical version string looks like 'OpenSSL 1.0.1e-fips 11 Feb 2013'
+  const openSSLVersionPattern = /^([^\s]*)\s([0-9]+\.[0-9]+)/;
+  const match = openSSLVersionPattern.exec(version);
+
+  if (match === null) {
+    throw new Error('No version of OpenSSL was found. Volta requires a valid version of OpenSSL.');
```

Comment:
Instead of throwing, should we default to `1.1` (basically what we were doing before)? What do you think?

One thing to tweak if we do keep this error, we should update the error message to mention `volta-cli/action` (instead of `"Volta"`) and we should also include the raw output from `openssl version`.

Reply:
The issue with defaulting to 1.1 is that it won't be present on the system, since if we return a null match it means it explicitly _wasn't_ present. I think throwing makes more sense in this case, which will allow us to be more specific with the error for end users. Thoughts?

Reply:
Updated to include those changes you specified.